numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -156,7 +139,7 @@ def get_nvrtc_dso_path():
         )
         if lib_dir and os.path.exists(lib_dir):
             try:
-                major = detect_nvrtc_major_cuda_version(lib_dir)
+                major = get_major_cuda_version(lib_dir)
```

Comment:
You can't do this because you could be running a newer CUDA driver with older library versions. I.E. I could run a CUDA 12 or CUDA 13 or CUDA 14 driver version with CUDA 11 libraries.

If we can load the cuda runtime (assuming it's a dependency already?) we could use the major version from that.